### PR TITLE
Use python scripts for CI instead of bash

### PR DESCRIPTION
**Pull request by nvaytet**
- Replace `make_and_install.*` with `build_cpp.py` for building C++ library and tests
- Replace `pre_docs.sh` and sphinx command with `docs/make_docs.py` script
- Use `git diff --exit-code` for brevity

See companion PRs in `scipp` (https://github.com/scipp/scipp/pull/1945) and `scippneutron` (https://github.com/scipp/scippneutron/pull/80)

### CODE REVIEW

File: code_quality_checks.yml [hunk 1]
```diff
@@ -46,17 +45,15 @@ stages:
               set -ex
               cmake-format --version
               # Perform format checking on all CMake files
-              find . -type f -name CMakeLists.txt -or -name '*.cmake' | xargs cmake-format --check
+              find . -type f -name CMakeLists.txt -or -name '*.cmake' | xargs cmake-format -i
+              git diff --exit-code
```
Comment (SimonHeybrock) [inline code_quality_checks.yml:49]: Will this now print the diff if formatting is wrong? Nice!

**Review by SimonHeybrock — APPROVED** [review]
Looks good.

File: code_quality_checks.yml (hunk 1 above)
Comment (nvaytet) [reply code_quality_checks.yml:49]: Yes it will

**Review by OwenArnold — APPROVED** [review]
Looks good.

How easy will it be to debug this? I'm thinking of failures in the subprocesses started. Do we see what's happened?

Also, given the above I wonder if we ought to add more protection on the python side. For example testing that all paths provided exist before proceeding further. Might same time later in maintenance.

**Comment by nvaytet** [comment]
> How easy will it be to debug this? I'm thinking of failures in the subprocesses started. Do we see what's happened?

You can for example here: https://dev.azure.com/scipp/pipelines/_build/results?buildId=5864&view=logs&j=a111e0f8-30a5-5221-2818-2887bca8cba8&t=41395b0a-808e-5913-0527-8eeeeee9b633&l=354
An error in the test executable name

**Comment by nvaytet** [comment]
you can also see here in the case of a compilation error: https://dev.azure.com/scipp/pipelines/_build/results?buildId=5871&view=logs&j=a111e0f8-30a5-5221-2818-2887bca8cba8&t=41395b0a-808e-5913-0527-8eeeeee9b633

**Comment by nvaytet** [comment]
> For example testing that all paths provided exist before proceeding further.

Which parts are you talking about in particular?

**Comment by OwenArnold** [comment]
> Which parts are you talking about in particular?

There are a number of places for example (1) where you construct a path to the OSX SDK. (2) ahead of the calls to move.

**Pull request merged**